src/DropDownMenu/DropDownMenu.jsx

```diff
@@ -0,0 +1,379 @@
+import warning from 'warning';
+
+const DropDownMenu = React.createClass({
+
+  mixins: [StylePropable],
+
+  contextTypes: {
+    muiTheme: React.PropTypes.object,
+  },
+
+  //for passing default theme context to children
+  childContextTypes: {
+    muiTheme: React.PropTypes.object,
+  },
+
+  getChildContext() {
+    return {
+      muiTheme: this.state.muiTheme,
+    };
+  },
+
+  // The nested styles for drop-down-menu are modified by toolbar and possibly
+  // other user components, so it will give full access to its js styles rather
+  // than just the parent.
+  propTypes: {
+    autoWidth: React.PropTypes.bool,
+    children: React.PropTypes.node,
+    className: React.PropTypes.string,
+    disabled: React.PropTypes.bool,
+    displayMember: React.PropTypes.string, // DEPRECATE
+    iconStyle: React.PropTypes.object,
+    labelMember: React.PropTypes.string, // DEPRECATE
+    labelStyle: React.PropTypes.object, // DEPRECATE
+    maxHeight: React.PropTypes.number,
+    menuItems: React.PropTypes.array, // DEPRECATE
+    menuStyle: React.PropTypes.object,
+    onChange: React.PropTypes.func,
+    openImmediately: React.PropTypes.bool,
+    selectedIndex: React.PropTypes.number, // DEPRECATE
+    style: React.PropTypes.object,
+    underlineStyle: React.PropTypes.object,
+    value: React.PropTypes.any,
+    valueLink: React.PropTypes.object, // DEPRECATE
+    valueMember: React.PropTypes.string, // DEPRECATE
+  },
+
+  getDefaultProps() {
+    return {
+      autoWidth: true,
+      disabled: false,
+      valueMember: 'payload',
+      displayMember: 'text',
+      openImmediately: false,
+      maxHeight: 500,
+      labelMember: 'text',
+    };
+  },
+
+  getInitialState() {
+    this._testDeprecations();
+    return {
+      open: this.props.openImmediately,
+      selectedIndex: this._isControlled() ? null : (this.props.selectedIndex || 0),
+      muiTheme: this.context.muiTheme ? this.context.muiTheme : ThemeManager.getMuiTheme(DefaultRawTheme),
+    };
+  },
+
+  componentDidMount() {
+    if (this.props.autoWidth) this._setWidth();
+    if (this.props.hasOwnProperty('selectedIndex')) this._setSelectedIndex(this.props.selectedIndex);
+  },
+
+  componentWillReceiveProps(nextProps, nextContext) {
+    const newMuiTheme = nextContext.muiTheme ? nextContext.muiTheme : this.state.muiTheme;
+    this.setState({muiTheme: newMuiTheme});
+
+    if (this.props.autoWidth) this._setWidth();
+    if (nextProps.hasOwnProperty('value') || nextProps.hasOwnProperty('valueLink')) {
+      return;
+    }
+    else if (nextProps.hasOwnProperty('selectedIndex')) {
+      this._setSelectedIndex(nextProps.selectedIndex);
+    }
+  },
+
+  getStyles() {
+    const {disabled} = this.props;
+    const spacing = this.state.muiTheme.rawTheme.spacing;
+    const palette = this.state.muiTheme.rawTheme.palette;
+    const accentColor = this.state.muiTheme.dropDownMenu.accentColor;
+    return {
+      root: {
+        transition: Transitions.easeOut(),
```

Comment:
Even though we're not currently or able to lint it, I've been trying to alphabetically order the keys in our style objects as well. Do you think that is worth while?